atomicapp/providers/openshift.py

```diff
@@ -133,6 +146,49 @@ def process_template(self, url, template):
             logger.error(msg)
             raise ProviderFailedException(msg)
 
+    def execute(self, namespace, pod, container, command):
+        """
+        Execute a command in a container in an Openshift pod.
+
+        Args:
+            namespace (str): Namespace
+            pod (str): Pod name
+            container (str): Container name inside pod
+            command (str): Command to execute
+
+        Returns:
+            Command output (str)
+        """
+        args = {
+            'token': self.access_token,
+            'namespace': namespace,
+            'pod': pod,
+            'container': container,
+            'command': ''.join(['command={}&'.format(word) for word in command.split()])
+        }
+        url = 'wss://' + urljoin(
+            self.kubernetes_api,
+            'namespaces/{namespace}/pods/{pod}/exec?'
+            'access_token={token}&container={container}&'
+            '{command}stdout=1&stdin=0&tty=0'.format(
+              **args)).split('://', 1)[-1]
+
+        results = []
+
+        ws = websocket.WebSocketApp(
+            url,
+            on_message=lambda ws, message: self._handle_exec_reply(ws, message, results))
+
+        ws.run_forever(sslopt={'cert_reqs': ssl.CERT_NONE})
+
+        return ''.join(results)
+
+    def _handle_exec_reply(self, ws, message, results):
+        """
+        Handle reply message for exec call
+        """
+        results.append(message[1:])
```

Comment:
Comment on `[1:]`. I still don't know why we need this but we should at comment. Either we should figure out why it is needed and state why or we should just comment that we don't know why.